src/tts_text/borger_dk.py

```diff
@@ -0,0 +1,269 @@
+
+    # Get the overall categories from the front page
+    # retry if it fails
+    soup: BeautifulSoup = BeautifulSoup("")
+    while not soup.contents:
+        soup = get_soup(url=BASE_URL, dynamic=True)
+
+    category_urls = [
+        BASE_URL + url_suffix.contents[1].attrs["href"]
+        for url_suffix in soup.find_all("li", class_="col-12")
+    ]
+
+    # Extract all articles
+    all_articles: list[str] = list()
+    found_urls: list[str] = list()
+    desc = "Extracting articles from borger.dk"
+    for category_url in tqdm(category_urls, desc=desc, leave=True):
+        # Borger.dk is structured in to categories, which are structured in to
+        # subcategories, which are structured in to articles. Every article contains
+        # a text, and a link to another article. The links go both up, down and
+        # sideways in the hierarchy, so we need to restrict ourselves to one category
+        # at a time, and only follow links downwards, otherwise the recursion will
+        # never end. This is why we use the category variable.
+        category = category_url.split("/")[-1]
+        category_articles, found_urls = extract_all_articles(
+            cfg=cfg, url=category_url, category=category, found_urls=found_urls
+        )
+        all_articles.extend(category_articles)
+
+    # Split the articles into sentences
+    dataset = extract_sentences(
+        corpus=all_articles, min_sentence_length=cfg.min_sentence_length
+    )
+
+    # Ensure sentences end with appropriate punctuation
+    dataset = [
+        sentence + "." if re.match(r".*[.?!]$", sentence) is None else sentence
+        for sentence in dataset
+    ]
+
+    # Remove sentences not starting with a capital letter
+    dataset = [sentence for sentence in dataset if sentence[0].isupper()]
+
+    # Save the dataset
+    with dataset_path.open("w", encoding="utf-8") as f:
+        f.write("\n".join(dataset))
+
+    return dataset
+
+
+def extract_all_articles(
+    cfg: DictConfig, url: str, category: str, parsed_urls: list[str] = [], found_urls=[]
+) -> tuple[list[str], list[str]]:
+    """Extract all articles from a category page.
+
+    These pages have arbitrarily nested subcategories, so this function is called
+    recursively. Each recursion checks the sidebar for new subcategories, and if none
+    are found, then the page is assumed to contain an article. The article is
+    extracted, and the links in the article are followed and extracted as well.
+    The recursion ends when no new subcategories are found, and the articles are
+    either only text, or theirs links are not suitable for further recursion. See
+    `get_suitable_links` for the criteria for suitable links.
+
+    Args:
+        cfg: The Hydra configuration object.
+        category: The category of the page.
+        parsed_urls: A list of URLs that have already been parsed.
+        found_urls: A list of URLs that have already been found, but not necessarily
+            parsed.
```

Comment:
If any of the descriptions are longer than a single line, then they all need to be formatted with "hanging indent", as described in [the Google style guide](https://google.github.io/styleguide/pyguide.html#doc-function-args):

> If the description is too long to fit on a single 80-character line, use a hanging indent of 2 or 4 spaces more than the parameter name (be consistent with the rest of the docstrings in the file).

They require that this choice (hanging indent or no hanging indent) stays consistent throughout each file, but to be a bit lax we could say that we at the very least we should be consistent within each function 🙂 

```suggestion
        cfg: 
            The Hydra configuration object.
        category: 
            The category of the page.
        parsed_urls: 
            A list of URLs that have already been parsed.
        found_urls: 
            A list of URLs that have already been found, but not necessarily parsed.
```